enterprise/metrics/src/main/java/org/neo4j/metrics/output/CsvReporterSingle.java

```diff
@@ -0,0 +1,413 @@
+         * @param durationUnit a unit of time
+         * @return {@code this}
+         */
+        public Builder convertDurationsTo( TimeUnit durationUnit )
+        {
+            this.durationUnit = durationUnit;
+            return this;
+        }
+
+        /**
+         * Use the given {@link Clock} instance for the time.
+         *
+         * @param clock a {@link Clock} instance
+         * @return {@code this}
+         */
+        public Builder withClock( Clock clock )
+        {
+            this.clock = clock;
+            return this;
+        }
+
+        /**
+         * Use the given {@link Log} for reporting any errors.
+         *
+         * @param logger A Log instance, never null.
+         * @return {@code this}
+         */
+        public Builder withLogger( Log logger )
+        {
+            this.logger = logger;
+            return this;
+        }
+
+        /**
+         * Only report metrics which match the given filter.
+         *
+         * @param filter a {@link MetricFilter}
+         * @return {@code this}
+         */
+        public Builder filter( MetricFilter filter )
+        {
+            this.filter = filter;
+            return this;
+        }
+
+        /**
+         * Builds a {@link CsvReporterSingle} with the given properties, writing {@code .csv} data to the
+         * given file.
+         *
+         * @param file the file in which the {@code .csv} data will be inserted
+         * @return a {@link CsvReporterSingle}
+         */
+        public CsvReporterSingle build( File file )
+        {
+            return new CsvReporterSingle( registry,
+                    file,
+                    rateUnit,
+                    durationUnit,
+                    clock,
+                    logger,
+                    filter );
+        }
+    }
+
+    private static final ThreadLocal<SimpleDateFormat> ISO8601 = new ThreadLocal<SimpleDateFormat>()
+    {
+        @Override
+        protected SimpleDateFormat initialValue()
+        {
+            return new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss" );
+        }
+    };
+
+    private final File file;
+    private final Clock clock;
+    private final Log logger;
+    private PrintWriter out;
+
+    private CsvReporterSingle( MetricRegistry registry,
+                               File file,
+                               TimeUnit rateUnit,
+                               TimeUnit durationUnit,
+                               Clock clock,
+                               Log logger,
+                               MetricFilter filter )
+    {
+        super( registry, "csv-reporter-single", filter, rateUnit, durationUnit );
+        this.file = file;
+        this.clock = clock;
+        this.logger = logger;
+    }
+
+    @Override
+    public void stop()
+    {
+        super.stop();
+
+        if ( out != null )
+        {
+            out.close();
+        }
+    }
+
+    @Override
+    public void report( SortedMap<String,Gauge> gauges,
```

Comment:
About which flag we talking about here?
Lets imagine that we are in `report` method in Thread1 and somebody stop the server and that is in another thread `Thread2`. Now we have a possibility to have closed out in the middle of report call, as example of possible problems? or something preventing that?